### ISSUE TITLE
Programmatically way to determine if paho-mqtt processes queue is empty?

### ISSUE THREAD

**Issue by Jibun-no-Kage** [eclipse/paho.mqtt.python#865]
Programmatically way to determine if paho-mqtt processes queue completed?  For example, if I call client.disconnect(), the on_disconnect call back may need a bit more time to complete, is there a way I can test for this?

**Comment by PierreF** [comment]
Hi,

no I don't think there is such feature currently. I agree it seems a valuable feature.

Depending on your use-case you might track per-message with [wait_for_publish](https://eclipse.dev/paho/files/paho.mqtt.python/html/client.html#paho.mqtt.client.MQTTMessageInfo.wait_for_publish), or if the number of message is too large just using on_publish callback (like count number of published message, if it match number of emitted message, the queue is empty).

**Comment by Jibun-no-Kage** [comment]
Yes... using on_publish method would seem to be possible in my use case.  Is there a process to request such a feature be added?  Seems straight-forward to add a small property or method to just report the queue length?
